### ISSUE TITLE
Rake aborted

### ISSUE THREAD

**Issue by ghost** [danbooru/danbooru#4214]
I used the install.sh file. 
Then I ran bundle install (btw you need to run "apt install -y build-essential patch ruby-dev zlib1g-dev liblzma-dev" for that to even work, was never mentioned in the docs or in the install script), Then I run "bundle exec rake db:create db:migrate" and get a rake aborted error with bootsnap and activesupport. 

Btw I'm using Ubuntu Server 18.04

**Comment by evazion** [comment]
Please copy/paste the actual command output. My first guess is that Postgres isn't installed or running. Try doing `psql --list`. Also try `bundle exec rails routes` to check that Rails can run.

**Comment by thenamewastaken** [comment]
I met a similar problem, in that rake aborted with errors from bootsnap and a bunch of other programs.
In my case the error was "uninitialized constant Danbooru::Configuration::Aws"

**Comment by evazion** [comment]
Please post the full output. My guess is that it couldn't find the AWS library because the dependencies weren't installed. Try doing `bundle install` first.

**Comment by thenamewastaken** [comment]
I did run `bundle install` (multiple times actually, just to make sure), and the problem persists.
I think the problem is that ruby thinks `Aws` is a module within Danbooru's configuration rather than a gem.
Full log:
```
/var/www/danbooru2/shared/bundle/ruby/2.6.0/gems/bootsnap-1.4.5/lib/bootsnap/load_path_cache/core_ext/active_support.rb:60:in `block in load_missing_constant': uninitialized constant Danbooru::Configuration::Aws (NameError)
	from /var/www/danbooru2/shared/bundle/ruby/2.6.0/gems/bootsnap-1.4.5/lib/bootsnap/load_path_cache/core_ext/active_support.rb:16:in `allow_bootsnap_retry'
	from /var/www/danbooru2/shared/bundle/ruby/2.6.0/gems/bootsnap-1.4.5/lib/bootsnap/load_path_cache/core_ext/active_support.rb:59:in `load_missing_constant'
	from /var/www/danbooru2/releases/20191126210547/config/danbooru_default_config.rb:564:in `aws_credentials'
	from /var/www/danbooru2/releases/20191126210547/config/danbooru_default_config.rb:665:in `method_missing'
	from /var/www/danbooru2/releases/20191126210547/app/logical/storage_manager/s3.rb:5:in `<class:S3>'
	from /var/www/danbooru2/releases/20191126210547/app/logical/storage_manager/s3.rb:1:in `<main>'
	from /var/www/danbooru2/shared/bundle/ruby/2.6.0/gems/bootsnap-1.4.5/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:22:in `require'
	from /var/www/danbooru2/shared/bundle/ruby/2.6.0/gems/bootsnap-1.4.5/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:22:in `block in require_with_bootsnap_lfi'
	from /var/www/danbooru2/shared/bundle/ruby/2.6.0/gems/bootsnap-1.4.5/lib/bootsnap/load_path_cache/loaded_features_index.rb:92:in `register'
	from /var/www/danbooru2/shared/bundle/ruby/2.6.0/gems/bootsnap-1.4.5/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:21:in `require_with_bootsnap_lfi'
	from /var/www/danbooru2/shared/bundle/ruby/2.6.0/gems/bootsnap-1.4.5/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:30:in `require'
	from /var/www/danbooru2/shared/bundle/ruby/2.6.0/gems/activesupport-6.0.1/lib/active_support/dependencies/interlock.rb:14:in `block in loading'
	from /var/www/danbooru2/shared/bundle/ruby/2.6.0/gems/activesupport-6.0.1/lib/active_support/concurrency/share_lock.rb:151:in `exclusive'
	from /var/www/danbooru2/shared/bundle/ruby/2.6.0/gems/activesupport-6.0.1/lib/active_support/dependencies/interlock.rb:13:in `loading'
	from /var/www/danbooru2/shared/bundle/ruby/2.6.0/gems/bootsnap-1.4.5/lib/bootsnap/load_path_cache/core_ext/active_support.rb:48:in `block in require_or_load'
	from /var/www/danbooru2/shared/bundle/ruby/2.6.0/gems/bootsnap-1.4.5/lib/bootsnap/l
 DEBUG [35962dd6] 	oad_path_cache/core_ext/active_support.rb:16:in `allow_bootsnap_retry'
	from /var/www/danbooru2/shared/bundle/ruby/2.6.0/gems/bootsnap-1.4.5/lib/bootsnap/load_path_cache/core_ext/active_support.rb:47:in `require_or_load'
	from /var/www/danbooru2/shared/bundle/ruby/2.6.0/gems/bootsnap-1.4.5/lib/bootsnap/load_path_cache/core_ext/active_support.rb:85:in `depend_on'
	from /var/www/danbooru2/shared/bundle/ruby/2.6.0/gems/railties-6.0.1/lib/rails/engine.rb:480:in `block (2 levels) in eager_load!'
	from /var/www/danbooru2/shared/bundle/ruby/2.6.0/gems/railties-6.0.1/lib/rails/engine.rb:479:in `each'
	from /var/www/danbooru2/shared/bundle/ruby/2.6.0/gems/railties-6.0.1/lib/rails/engine.rb:479:in `block in eager_load!'
	from /var/www/danbooru2/shared/bundle/ruby/2.6.0/gems/railties-6.0.1/lib/rails/engine.rb:476:in `each'
	from /var/www/danbooru2/shared/bundle/ruby/2.6.0/gems/railties-6.0.1/lib/rails/engine.rb:476:in `eager_load!'
	from /var/www/danbooru2/shared/bundle/ruby/2.6.0/gems/railties-6.0.1/lib/rails/engine.rb:356:in `eager_load!'
	from /var/www/danbooru2/shared/bundle/ruby/2.6.0/gems/railties-6.0.1/lib/rails/application/finisher.rb:123:in `each'
	from /var/www/danbooru2/shared/bundle/ruby/2.6.0/gems/railties-6.0.1/lib/rails/application/finisher.rb:123:in `block in <module:Finisher>'
	from /var/www/danbooru2/shared/bundle/ruby/2.6.0/gems/railties-6.0.1/lib/rails/initializable.rb:32:in `instance_exec'
	from /var/www/danbooru2/shared/bundle/ruby/2.6.0/gems/railties-6.0.1/lib/rails/initializable.rb:32:in `run'
	from /var/www/danbooru2/shared/bundle/ruby/2.6.0/gems/railties-6.0.1/lib/rails/initializable.rb:61:in `block in run_initializers'
	from /home/vanni/.rbenv/versions/2.6.3/lib/ruby/2.6.0/tsort.rb:228:in `block in tsort_each'
	from /home/vanni/.rbenv/versions/2.6.3/lib/ruby/2.6.0/tsort.rb:350:in `block (2 levels) in each_strongly_connected_component'
	from /home/vanni/.rbenv/versions/2.6.3/lib/ruby/2.6.0/tsort.rb:431:in `each_strongly_connected_component_from'
	from /home/vanni/.rbenv/versions/2.6.3/lib/ruby/2.6.0/tsort.rb:349:in `block in each_strongly_connected_component'
	from /home/vanni/.rbenv/versions/2.6.3/lib/ruby/2.6.0/tsort.rb:347:in `each'
	from /home/vanni/.rbenv/versions/2.6.3/lib/ruby/2.6.0/tsort.rb:347:in `call'
	from /home/vanni/.rbenv/versions/2.6.3/lib/ruby/2.6.0/tsort.rb:347:in `each_strongly_connected_component'
	from /home/vanni/.rbenv/versions/2.6.3/lib/ruby/2.6.0/tsort.rb:226:in `tsort_each'
	from /home/vanni/.rbenv/versions/2.6.3/lib/ruby/2.6.0/tsort.rb:205:in `tsort_each'
	from /var/www/danbooru2/shared/bundle/ruby/2.6.0/gems/railties-6.0.1/lib/rails/initializable.rb:60:in `run_initializers'
	from /var/www/danbooru2/shared/bundle/ruby/2.6.0/gems/railties-6.0.1/lib/rails/application.rb:363:in `initialize!'
	from /var/www/danbooru2/releases/20191126210547/config/environment.rb:7:in `<main>'
	from /var/www/danbooru2/shared/bundle/ruby/2.6.0/gems/bootsnap-1.4.5/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:22:in `require'
	from /var/www/danbooru2/shared/bundle/ruby/2.6.0/gems/bootsnap-1.4.5/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:22:in `block in require_with_bootsnap_lfi'
	from /var/www/danbooru2/shared/bundle/ruby/2.6.0/gems/bootsnap-1.4.5/lib/bootsnap/load_path_cache/loaded_features_index.rb:92:in `register'
	from /var/www/danbooru2/shared/bundle/ruby/2.6.0/gems/bootsnap-1.4.5/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:21:in `require_with_bootsnap_lfi'
	from /var/www/danbooru2/shared/bundle/ruby/2.6.0/gems/bootsnap-1.4.5/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:30:in `require'
	from /var/www/danbooru2/shared/bundle/ruby/2.6.0/gems/railties-6.0.1/lib/rails/application.rb:339:in `require_environment!'
	from /var/www/danbooru2/shared/bundle/ruby/2.6.0/gems/railties-6.0.1/lib/rails/command/actions.rb:28:in `require_environment!'
	from /var/www/danbooru2/shared/bundle/ruby/2.6.0/gems/railties-6.0.1/lib/rails/command/actions.rb:15:in `require_application_and_environment!'
	from /var/www/danbooru2/shared/bundle/ruby/2.6.0/gems/railties-6.0.1/lib/rails/commands/runner/runner_command.rb:33:in `perform'
	from /var/www/danbooru2/shared/bundle/ruby/2.6.0/gems/thor-0.20.3/lib/thor/command.rb:27:in `run'
	from /var/www/danbooru2/shared/bundle/ruby/2.6.0/gems/thor-0.20.3/lib/thor/invocation.rb:126:in `invoke_command'
	from /var/www/danbooru2/shared/bundle/ruby/2.6.0/gems/thor-0.20.3/lib/thor.rb:387:in `dispatch'
	from /var/www/danbooru2/shared/bundle/ruby/2.6.0/gems/railties-6.0.1/lib/rails/command/base.rb:65:in `perform'
	from /var/www/danbooru2/shared/bundle/ruby/2.6.0/gems/railties-6.0.1/lib/rails/command.rb:46:in `invoke'
	from /var/www/danbooru2/shared/bundle/ruby/2.6.0/gems/railties-6.0.1/lib/rails/commands.rb:18:in `<main>'
	from /var/www/danbooru2/shared/bundle/ruby/2.6.0/gems/bootsnap-1.4.5/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:22:in `require'
	from /var/www/danbooru2/shared/bundle/ruby/2.6.0/gems/bootsnap-1.4.5/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:22:in `block in require_with_bootsnap_lfi'
	from /var/www/danbooru2/shared/bundle/ruby/2.6.0/gems/bootsnap-1.4.5/lib/bootsnap/load_path_cache/loaded_features_index.rb:92:in `register'
	from /var/www/danbooru2/shared/bundle/ruby/2.6.0/gems/bootsnap-1.4.5/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:21:in `require_with_bootsnap_lfi'
	from /var/www/danbooru2/shared/bundle/ruby/2.6.0/gems/bootsnap-1.4.5/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:30:in `require'
	from bin/rails:4:in `<main>'
```

**Comment by evazion** [comment]
Possibly fixed in 6382aec85e67be0c5afe211bf20a92b0853b337f. This switches to zeitwerk, the new code autoloader in Rails 6 that supposedly fixes various problems with autoloading like this. I couldn't reproduce this bug so I'm not sure if this actually fixes it.

https://github.com/fxn/zeitwerk
https://guides.rubyonrails.org/autoloading_and_reloading_constants.html
https://guides.rubyonrails.org/v5.2/autoloading_and_reloading_constants.html#common-gotchas

**Comment by thenamewastaken** [comment]
Seems to be fixed on my end.
